### PR TITLE
Personr connector

### PR DESCRIPTION
**Pull request by charliewesterman**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [Y] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [Y] I attest that the connector works and I verified by deploying and testing all the operations. 
- [Y] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [Y] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [Y] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [Y] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

### CODE REVIEW

**Comment by charliewesterman** [comment]
@microsoft-github-policy-service agree company="Personr"

File: certified-connectors/personr/readme.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Personr Connector
+
+The Personr connector provides access to a simple, powerful REST API. Using this API, you can create, verify and request the status or more details of an applicant.
```
Comment (vmanoharas) [inline certified-connectors/personr/readme.md:2]: Please update readme.md file as per the template in the below link, please ensure that all the required fields added/updated duly. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md

File: certified-connectors/personr/personr.swagger.json [hunk 2]
```diff
@@ -0,0 +1,487 @@
+{
+  "swagger": "2.0",
```
Comment (vmanoharas) [inline certified-connectors/personr/personr.swagger.json]: Please modify swagger filename as per the templates mentioned in this path. 
https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json

File: certified-connectors/personr/personr.swagger.json [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+  "info": {
+    "version": "1.0.0",
+    "title": "Personr API",
+    "description": "Create, verify and retrieve applicant data using Personr. Effortlessly comply with KYC/AML, using PowerAutomate to integrate into your own technology."
+  },
```
Comment (vmanoharas) [inline certified-connectors/personr/personr.swagger.json]: Please add contact information, as per the template
https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json

Additionally, please refer the swagger file schema and update duly. 
https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/apiDefinition.swagger.schema.json

File: certified-connectors/personr/apiProperties.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "properties": {
+    "capabilities": [],
```
Comment (vmanoharas) [inline certified-connectors/personr/apiProperties.json]: Please update apiProperties.json file as per the below link. 
https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiProperties.json

**Review by vmanoharas — COMMENTED** [review]
Hello @charliewesterman,
Kindly resolve review comments.

**Comment by charliewesterman** [comment]
Hi @vmanoharas vmanoharas, I have just updated all of the required changes now. Thanks.

**Comment by vmanoharas** [comment]
Hi @charliewesterman 

Thank you very much for resolving all the comments.  Couple you please resolve api property validation errors.  Please refer the schema file in this path. 
https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/paconn-apiProperties.schema.json

**Comment by charliewesterman** [comment]
> Hi @charliewesterman
> 
> Thank you very much for resolving all the comments. Couple you please resolve api property validation errors. Please refer the schema file in this path. https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/paconn-apiProperties.schema.json

Hi @vmanoharas - I have changed the apiProperties file to this recommendation from Microsoft: https://learn.microsoft.com/en-us/connectors/custom-connectors/paconn-cli

File: certified-connectors/personr/personr.swagger.json [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+      "post": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "Upload Documents to an Applicant",
+        "operationId": "Api-document-upload",
+        "description": "Use this action to upload documents to an applicant, instead of sending them a link. You can only send one document at a time, when all required documents are uploaded the applicant will move to the pending status.  If you're uploading double-side documents, make sure to send FRONT_SIDE and BACK_SIDE in the idDocSubType parameter. When both sides are present and valid the verification step will be considered complete. Images must be uploaded in base64 format.",
```
Comment (vmanoharas) [inline certified-connectors/personr/personr.swagger.json]: Hello @charliewesterman, 

please keep all operations descriptions short to fit in the text box. please see our [docs](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations).

File: certified-connectors/personr/personr.swagger.json [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+              }
+            }
+          }
+        ]
+      }
+    },
+    "/api-applicant-status": {
+      "post": {
+        "summary": "Retrieve Applicant Status",
+        "description": "Use this action to retrieve an applicant verification status. If the verification has been already reviewed, the response will contain data with more details about the result.",
```
Comment (vmanoharas) [inline certified-connectors/personr/personr.swagger.json]: please keep all operations descriptions short to fit in the text box. please see our [docs](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations).

File: certified-connectors/personr/personr.swagger.json [hunk 7]
```diff
@@ -0,0 +1,491 @@
...
+    "/api-request-applicant-check": {
+      "post": {
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        },
+        "summary": "Request Applicant Check",
+        "description": "Use this action to let us know you've finished uploading documents to an applicant, and to start the verification process. Note: only use this if you are uploading documents. Verifications using a link don't need to use this action.",
```
Comment (vmanoharas) [inline certified-connectors/personr/personr.swagger.json]: please keep all operations descriptions short to fit in the text box. please see our [docs](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations).

**Review by vmanoharas — COMMENTED** [review]
Hello @charliewesterman, 

thank you for fixing the errors.  Please resolve review comments.  

Additionally, please change file names as mentioned in the below link. 

https://github.com/microsoft/PowerPlatformConnectors/tree/dev/templates/certified-connectors

**Comment by charliewesterman** [comment]
Hi @vmanoharas - thanks for your prompt reply. 

I have renamed the files according to your link, and shortened the descriptions.

**Comment by charliewesterman** [comment]
Hi @vmanoharas - any updates?

**Comment by vmanoharas** [comment]
Hello @charliewesterman, 

Please accept my sincere apology for the delay, will review now.

**Comment by vmanoharas** [comment]
Hello @charliewesterman, 

The swagger validation is failing, can you please sync your repository with main repo. Then, reopen the PR again with latest commit with the changes

**Comment by charliewesterman** [comment]
@vmanoharas just updated the swagger file, as there was an error - my apologies.

File: certified-connectors/personr/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,492 @@
+{
+    "swagger": "2.0",
+    "info": {
+        "version": "1.0.0",
+        "title": "Personr",
+        "description": "Create, verify and retrieve applicant data using Personr. Effortlessly comply with KYC/AML, using PowerAutomate to integrate into your own technology.",
+        "contact": {
```
Comment (vmanoharas) [inline certified-connectors/personr/apiDefinition.swagger.json:7]: Hello @charliewesterman, 

The PowerAutomate word cannot be used, could you please remove this, as this is restricted word.

File: certified-connectors/personr/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,492 @@
...
+            "name": "Authorization"
+        }
+    },
+    "security": [
+        {
+            "Auth. Token": []
+        }
+    ],
+    "tags": []
+}
```
Comment (vmanoharas) [inline certified-connectors/personr/apiDefinition.swagger.json:506]: "x-ms-connector-metadata" this tag is missing, could you please add the same here, please refer the template of swagger.json file. https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json

**Review by vmanoharas — COMMENTED** [review]
Hello @charliewesterman, 

My apology, couple of comments, kindly fix them.

**Comment by charliewesterman** [comment]
No worries, thanks @vmanoharas - just updated again now.

File: certified-connectors/personr/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,506 @@
...
+            "propertyName": "Website",
+            "propertyValue": "https://personr.co"
+        },
+        {
+            "propertyName": "Privacy policy",
+            "propertyValue": "https://personr.co/legals"
+        },
+        {
+            "propertyName": "Categories",
+            "propertyValue": "Identity Verification"
```
Comment (vmanoharas) [inline certified-connectors/personr/apiDefinition.swagger.json]: Hello @charliewesterman, 

Only below categories are available, please update.  Please refer this URL - https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json#:~:text=%22x%2Dms%2Dconnector,%5D

[AI, Business Management, Business Intelligence, Collaboration, Commerce, Communication, Content and Files, Finance, Data, Human Resources, Internet of Things, IT Operations, Lifestyle and Entertainment, Marketing, Productivity, Sales and CRM, Security, Social Media, Website].

**Review by vmanoharas — COMMENTED** [review]
Hello @charliewesterman, 

Please update review comments.

**Comment by charliewesterman** [comment]
Thanks @vmanoharas, updated now.

**Review by vmanoharas — APPROVED** [review]
Hello @charliewesterman 
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Pull request merged**